tools/depcheck/pkg/cmd/pin.go

```diff
@@ -0,0 +1,134 @@
+
+	DecodedYamlBytes []byte
+
+	Out    io.Writer
+	ErrOut io.Writer
+}
+
+type PinImportsFlags struct {
+	LockFileName string
+	YamlFileName string
+}
+
+func (o *PinImportsFlags) ToOptions(out, errout io.Writer) (*PinImportsOpts, error) {
+	if len(o.LockFileName) == 0 || len(o.YamlFileName) == 0 {
+		return nil, fmt.Errorf("both a --glide.lock path and a --glide.yaml path must be specified")
+	}
+
+	yamlFileBytes, err := ioutil.ReadFile(o.YamlFileName)
+	if err != nil {
+		return nil, err
+	}
+	lockFileBytes, err := ioutil.ReadFile(o.LockFileName)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(yamlFileBytes) == 0 {
+		return nil, fmt.Errorf("--glide.yaml path contained an empty file")
+	}
+	if len(lockFileBytes) == 0 {
+		return nil, fmt.Errorf("--glide.lock path contained an empty file")
+	}
+
+	lockfile := &glide.LockFile{}
+	yamlfile := &glide.YamlFile{}
+	if err := lockfile.Decode(lockFileBytes); err != nil {
+		return nil, err
+	}
+	if err := yamlfile.Decode(yamlFileBytes); err != nil {
+		return nil, err
+	}
+
+	return &PinImportsOpts{
+		LockFile: lockfile,
+		YamlFile: yamlfile,
+
+		DecodedYamlBytes: yamlFileBytes,
+
+		Out:    out,
+		ErrOut: errout,
+	}, nil
+}
+
+func NewCmdPinImports(parent string, out, errout io.Writer) *cobra.Command {
+	flags := &PinImportsFlags{}
+
+	cmd := &cobra.Command{
+		Use:   "pin --glide.lock=FOO.lock --glide.yaml=FOO.yaml",
+		Short: "Updates all non-forked dependencies in a glide.yaml file to use the SHAs specified in a glide.lock file.",
+		Long: `Updates all non-forked dependencies in a glide.yaml file to use the SHAs specified in a glide.lock file.
+Any existing dependencies on a glide.yaml file containing a "repo" field are ignored.`,
+		Example: fmt.Sprintf(pinImportsExample, parent),
+		RunE: func(c *cobra.Command, args []string) error {
+			opts, err := flags.ToOptions(out, errout)
+			if err != nil {
+				return err
+			}
+
+			if err := opts.Validate(); err != nil {
+				return err
+			}
+
+			return opts.Run()
+		},
+	}
+
+	cmd.Flags().StringVar(&flags.LockFileName, "glide.lock", "glide.lock", "Path to glide.lock file used to indicate which dependencies should be updated or pinned to a specific revision in the glide.yaml file")
+	cmd.Flags().StringVar(&flags.YamlFileName, "glide.yaml", "glide.yaml", "Path to glide.yaml file used to pin dependencies from the given --glide.lock file")
+
+	return cmd
+}
+
+func (o *PinImportsOpts) Validate() error {
+	if o.YamlFile == nil || o.LockFile == nil {
+		return fmt.Errorf("both a glide.yaml and a glide.lock file are required")
+	}
+	if o.Out == nil || o.ErrOut == nil {
+		return fmt.Errorf("no output or error output buffer given")
+	}
+
+	return nil
+}
+
+func (o *PinImportsOpts) Run() error {
+	missing, warnings, err := glide.MissingImports(o.LockFile, o.YamlFile)
+	if err != nil {
+		return fmt.Errorf("error: unable to calculate missing imports: %v\n", err)
+	}
+
+	for _, w := range warnings {
+		fmt.Fprintf(o.ErrOut, w)
```

Comment:
This looks weird.  Shouldn't it be `fmt.Fprintf(o.ErrOut, "%s\n", w`?

Also, we don't pay extra for `for _, warning := warnings {`